kore/test/Test/SMT.hs

```diff
@@ -55,7 +55,7 @@ assertEqual'
 assertEqual' str expect = liftIO . assertEqual str expect
 
 runSMT :: SMT a -> IO a
```

Comment:
1. Extra `userInit` argument here.
2. Find uses of `declareSMTLemmas` in the tests. Use `runSMT` with `declareSMTLemmas` call as the extra argument.
3. Anywhere that we use `runSMT` without `declareSMTLemmas`, replace with `runNoSMT`.